src/renderer/html_handlebars/search.rs

```diff
@@ -0,0 +1,221 @@
+extern crate elasticlunr;
+
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::path::Path;
+
+use pulldown_cmark::*;
+use serde_json;
+use self::elasticlunr::Index;
+
+use book::BookItem;
+use config::Search;
+use errors::*;
+use renderer::RenderContext;
+use utils;
+use theme::searcher::Searcher;
+
+/// Creates all files required for search.
+pub fn create_files(ctx: &RenderContext) -> Result<()> {
+    let search_config = ctx.config.html_config().unwrap_or_default().search;
+    let destination = &ctx.destination;
+    let book = &ctx.book;
+
+    if !search_config.enable {
+        return Ok(());
+    }
+
+    let mut index = Index::new(&["title", "body", "breadcrumbs"]);
+
+    for item in book.iter() {
+        render_item(&mut index, &search_config, item)?;
+    }
+    
+    let json = write_to_json(index, &search_config)?;
+    let searcher = Searcher::new(&search_config.searcher);
+    utils::fs::write_file(destination, "searchindex.json", json.as_bytes())?;
+    utils::fs::write_file(destination, "searcher.js", &searcher.js)?;
+    utils::fs::write_file(destination, "mark.min.js", &searcher.mark_js)?;
+    utils::fs::write_file(destination, "elasticlunr.min.js", &searcher.elasticlunr_js)?;
+    debug!("Copying search files ✓");
+
+    Ok(())
+}
+
+fn make_doc_ref(anchor_base: &str, section_id: Option<String>) -> Cow<str> {
+    if let Some(id) = section_id {
+        format!("{}#{}", anchor_base, id).into()
+    } else {
+        anchor_base.into()
+    }
+}
+
+/// Renders markdown into flat unformatted text and adds it to the search index.
+fn render_item(index: &mut Index, search_config: &Search, item: &BookItem) -> Result<()> {
```

Comment:
Is it worth creating a bunch of tests for this so we can be sure the text gets stripped as intended?